lib/src/components/NxIconDropdown/stateful/__tests__/NxStatefulIconDropdown.test.tsx

```diff
@@ -5,187 +5,233 @@
+        toggleBtn = getByRole('button');
+
+    await user.click(toggleBtn);
+
+    const menu = container.querySelector('.nx-dropdown-menu'),
+        children = getByTestId('child');
+
+    expect(menu).toBeInTheDocument();
+
+    await act(async () => { await user.click(children); });
 
     expect(childClickSpy).toHaveBeenCalled();
-    expect(element!.find('.nx-dropdown-menu')).not.toExist();
+    expect(menu).not.toBeInTheDocument();
+  });
+
+  it('closes the dropdown when the Escape key is pressed and dropdown is open', async function() {
+    const user = userEvent.setup(),
+        { container, getByRole } = quickRender(),
+        toggleBtn = getByRole('button');
+
+    await user.click(toggleBtn);
+
+    expect(container.querySelector('.nx-dropdown-menu')).toBeInTheDocument();
+
+    toggleBtn.focus();
+    await user.keyboard('[Escape]');
+
+    expect(container.querySelector('.nx-dropdown-menu')).not.toBeInTheDocument();
+  });
+
+  it('does not open the dropdown when the Escape key is pressed and dropdown is closed', async function() {
+    const user = userEvent.setup(),
+        { container, getByRole } = quickRender(),
+        toggleBtn = getByRole('button');
+
+    expect(container.querySelector('.nx-dropdown-menu')).not.toBeInTheDocument();
 
-    act(() => {
-      element!.find('button.nx-icon-dropdown__toggle').getDOMNode().dispatchEvent(new MouseEvent('click', {
-        bubbles: true
-      }));
-    });
-    element!.update();
-    expect(element!.find('.nx-dropdown-menu')).toExist();
+    toggleBtn.focus();
+    await user.keyboard('[Escape]');
 
-    element!.unmount();
+    expect(container.querySelector('.nx-dropdown-menu')).not.toBeInTheDocument();
   });
+
+  it('does not close the dropdown when the Escape key is pressed and dropdown is disabled', async function() {
+    const user = userEvent.setup(),
+        { container, getByRole, rerender } = quickRender(),
+        toggleBtn = getByRole('button');
+
+    await user.click(toggleBtn);
+
+    expect(container.querySelector('.nx-dropdown-menu')).toBeInTheDocument();
+
+    rerender(<NxStatefulIconDropdown {...minimalProps} disabled={true}/>);
+
+    toggleBtn.focus();
+    await user.keyboard('[Escape]');
+
+    expect(container.querySelector('.nx-dropdown-menu')).toBeInTheDocument();
+  });
+
+  it('moves focus to the dropdown toggle button if a menu item is focused when the dropdown is closed',
+      async function() {
+        const props: Partial<Props> = {
+              children: <button data-testid="dropdown-button">Foo</button>
+            },
+            user = userEvent.setup(),
+            { getByTestId, getByRole } = quickRender(props),
+            toggleBtn = getByRole('button');
+
+        await user.click(toggleBtn);
+
+        const menuBtn = getByTestId('dropdown-button');
+
+        menuBtn.focus();
+
+        expect(document.activeElement).toBe(menuBtn);
+
+        await act(async () => { await user.click(toggleBtn); });
```

Comment:
Don't need `act`

Reply:
> Don't need act

Actually, using click here at all really isn't right.  Click will inherently move the focus to the clicked element.  We want to test that focus moves to the toggle button even despite that.  The main way I can think of to accomplish that - and the way I did it in `NxFilterDropdown` - is to use the ESC key rather than the mouse to close the dropdown while the `menuBtn` is focused.

Reply:
Make sense to me, updated here 2837602262ac9bf62d0739289962e6bc0b3be060, thanks.